### PR TITLE
Bug 1809511: block creation of indices with the write suffix for index management

### PR DESCRIPTION
**Pull request by jcantrill**
Block auto index creation for indices that end in `-write` to avoid a race condition between fluent writing logs and EO setting up index management

cc @QiaolingTang @ewolinetz 

https://bugzilla.redhat.com/show_bug.cgi?id=1809511

### CODE REVIEW

**Comment by ewolinetz** [comment]
based on the ES docs it lgtm.
is there any way we can write a test for this?

**Comment by jcantrill** [comment]
> based on the ES docs it lgtm.
> is there any way we can write a test for this?

I was wondering this myself.  I think this would be a good way to prove out some additional bits to IM.  Maybe try to post a doc....assert fail.  create IM....... try to post a doc again... assert success.  assert there is no index with suffix of 'write'

**Comment by ewolinetz** [comment]
I think for the sake of fixing the bug in a timely fashion, I'm okay with moving forward with just manually confirming first and add in unit testing as tech debt

**Comment by jcantrill** [comment]
@ewolinetz added task for tech debt to address in coming sprint https://issues.redhat.com/browse/LOG-663. I think there also needs to be logic added to re-index said index as well for cases where maybe the CLO upgrades before EO.  Please /lgtm

File: hack/testing/test-657-im-block-autocreate-for-write-suffix.sh [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  expect_success "oc -n ${TEST_NAMESPACE} create -f ${repo_dir}/hack/cr.yaml"
+
+  #wait for pod
+  wait_for_deployment_to_be_ready ${TEST_NAMESPACE} $(oc -n ${TEST_NAMESPACE} get deployment -l component=elasticsearch -o jsonpath={.items[0].metadata.name}) $((2 * $minute))
+
+fi
+
+pod=$(oc -n $TEST_NAMESPACE get pod -l component=elasticsearch -o jsonpath={.items[0].metadata.name})
+log::info Attempt to autocreate an index without a '-write' suffix...
+expect_success_and_text "oc -n $TEST_NAMESPACE exec $pod -c elasticsearch -- es_util --query=foo/_doc/1 -d {\"key\":\"value\"} -XPUT -w %{http_code}" ".*201"
```
Comment (ewolinetz) [inline hack/testing/test-657-im-block-autocreate-for-write-suffix.sh]: why is this a `201` and a success? wouldn't we expect a `200`?

File: hack/testing/test-657-im-block-autocreate-for-write-suffix.sh [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+log::info Attempt to autocreate an index with a '-write' suffix...
+expect_success_and_text "oc -n $TEST_NAMESPACE exec $pod -c elasticsearch -- es_util --query=foo-write/_doc/1 -d {\"key\":\"value\"} -XPUT -w %{http_code}" ".*404"
+
+log::info Explicitly creating an index with a '-write' suffix...
+expect_success_and_text "oc -n $TEST_NAMESPACE exec $pod -c elasticsearch -- es_util --query=foo-write -XPUT -w %{http_code}" ".*200"
+log::info Verifying can write to index with a '-write' suffix...
+expect_success_and_text "oc -n $TEST_NAMESPACE exec $pod -c elasticsearch -- es_util --query=foo-write/_doc/1 -d {\"key\":\"value\"} -XPUT  -w %{http_code}" ".*201"
```
Comment (ewolinetz) [inline hack/testing/test-657-im-block-autocreate-for-write-suffix.sh]: same question here

**Pull request merged**